rcl/test/rcl/test_log_level.cpp

```diff
@@ -0,0 +1,206 @@
+    }); \
+  }
+
+TEST(CLASSNAME(TestLogLevelFixture, RMW_IMPLEMENTATION), no_log_level) {
+  rcl_log_level_t * log_level = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(log_level, "process_name");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rcl_log_level_fini(log_level);
+  });
+  EXPECT_EQ(-1, log_level->default_log_level);
+  EXPECT_EQ(static_cast<size_t>(0), log_level->num_loggers);
+}
+
+TEST(CLASSNAME(TestLogLevelFixture, RMW_IMPLEMENTATION), default_log_level) {
+  rcl_log_level_t * log_level = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_level, "process_name", "--ros-args",
+    "--log-level", "debug");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rcl_log_level_fini(log_level);
+  });
+  EXPECT_EQ(static_cast<int>(RCUTILS_LOG_SEVERITY_DEBUG), log_level->default_log_level);
+  EXPECT_EQ(static_cast<size_t>(0), log_level->num_loggers);
+}
+
+TEST(CLASSNAME(TestLogLevelFixture, RMW_IMPLEMENTATION), logger_log_level_debug) {
+  rcl_log_level_t * log_level = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_level, "process_name", "--ros-args",
+    "--log-level", "rcl=debug");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rcl_log_level_fini(log_level);
+  });
+  EXPECT_EQ(-1, log_level->default_log_level);
+  EXPECT_EQ(static_cast<size_t>(1), log_level->num_loggers);
+  EXPECT_STREQ("rcl", log_level->logger_settings[0].name);
+  EXPECT_EQ(static_cast<int>(RCUTILS_LOG_SEVERITY_DEBUG), log_level->logger_settings[0].level);
+}
+
+TEST(CLASSNAME(TestLogLevelFixture, RMW_IMPLEMENTATION), logger_log_level_info) {
+  rcl_log_level_t * log_level = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_level, "process_name", "--ros-args",
+    "--log-level", "rcl=info");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rcl_log_level_fini(log_level);
+  });
+  EXPECT_EQ(-1, log_level->default_log_level);
+  EXPECT_EQ(static_cast<size_t>(1), log_level->num_loggers);
+  EXPECT_STREQ("rcl", log_level->logger_settings[0].name);
+  EXPECT_EQ(static_cast<int>(RCUTILS_LOG_SEVERITY_INFO), log_level->logger_settings[0].level);
+}
+
+TEST(CLASSNAME(TestLogLevelFixture, RMW_IMPLEMENTATION), default_log_level_with_logger) {
+  rcl_log_level_t * log_level = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_level, "process_name", "--ros-args",
+    "--log-level", "debug,rcl=debug");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rcl_log_level_fini(log_level);
+  });
+  EXPECT_EQ(static_cast<int>(RCUTILS_LOG_SEVERITY_DEBUG), log_level->default_log_level);
+  EXPECT_EQ(static_cast<size_t>(1), log_level->num_loggers);
+  EXPECT_STREQ("rcl", log_level->logger_settings[0].name);
+  EXPECT_EQ(static_cast<int>(RCUTILS_LOG_SEVERITY_DEBUG), log_level->logger_settings[0].level);
+}
+
+TEST(CLASSNAME(TestLogLevelFixture, RMW_IMPLEMENTATION), logger_with_default_log_level) {
+  rcl_log_level_t * log_level = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_level, "process_name", "--ros-args",
+    "--log-level", "rcl=debug,debug");
```

Comment:
I'm also unsure about `,` usage.
We could also allow multiple `--log-level` arguments, and use something like:
```
--ros-args --log-level info --log-level rcl:=debug --log-level rclcpp:=debug
```

of course, it's much more verbose. I'm not sure of what's the best option.